sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobBaseAPITest.groovy

```diff
@@ -0,0 +1,604 @@
+        thrown(IllegalArgumentException)
+
+        where:
+        input   | output   || _
+        true    | false    || _
+        false   | true     || _
+    }
+
+    @Unroll
+    def "Query AC"() {
+        setup:
+        match = setupBlobMatchCondition(bc, match)
+        leaseID = setupBlobLeaseCondition(bc, leaseID)
+        def bac = new BlobRequestConditions()
+            .setLeaseId(leaseID)
+            .setIfMatch(match)
+            .setIfNoneMatch(noneMatch)
+            .setIfModifiedSince(modified)
+            .setIfUnmodifiedSince(unmodified)
+        def expression = "SELECT * from BlobStorage"
+        BlobQueryOptions options = new BlobQueryOptions()
+            .setRequestConditions(bac)
+
+        when:
+        InputStream stream = bc.openQueryInputStream(expression, options)
+        stream.read()
+        stream.close()
+
+        then:
+        notThrown(BlobStorageException)
+
+        when:
+        bc.queryWithResponse(new ByteArrayOutputStream(), expression, options,null, null)
+
+        then:
+        notThrown(BlobStorageException)
+
+        where:
+        modified | unmodified | match        | noneMatch   | leaseID
+        null     | null       | null         | null        | null
+        oldDate  | null       | null         | null        | null
+        null     | newDate    | null         | null        | null
+        null     | null       | receivedEtag | null        | null
+        null     | null       | null         | garbageEtag | null
+        null     | null       | null         | null        | receivedLeaseID
+    }
+
+    @Unroll
+    def "Query AC fail"() {
+        setup:
+        setupBlobLeaseCondition(bc, leaseID)
+        def bac = new BlobRequestConditions()
+            .setLeaseId(leaseID)
+            .setIfMatch(match)
+            .setIfNoneMatch(setupBlobMatchCondition(bc, noneMatch))
+            .setIfModifiedSince(modified)
+            .setIfUnmodifiedSince(unmodified)
+        def expression = "SELECT * from BlobStorage"
+        BlobQueryOptions options = new BlobQueryOptions()
+            .setRequestConditions(bac)
+
+        when:
+        InputStream stream = bc.openQueryInputStream(expression, options)
+        stream.read()
+        stream.close()
+
+        then:
+        def e = thrown(IOException)
+        assert e.getCause() instanceof BlobStorageException
+
+        when:
+        bc.queryWithResponse(new ByteArrayOutputStream(), expression, options, null, null)
+
+        then:
+        thrown(BlobStorageException)
+
+        where:
+        modified | unmodified | match       | noneMatch    | leaseID
+        newDate  | null       | null        | null         | null
+        null     | oldDate    | null        | null         | null
+        null     | null       | garbageEtag | null         | null
+        null     | null       | null        | receivedEtag | null
+        null     | null       | null        | null         | garbageLeaseID
+    }
+}
```

Comment:
Do you test all the options like escape character and line delimiter? I don't remember seeing tests for that. Maybe you can make your upload data methods more generic to accept these options.